submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,88 @@
+
+Universal balancers can do what no balancer should be able to: no matter the configuration, they're input-balanced, output-balanced, *and* throughput-unlimited at the same time! This is a set of properties previously thought to be possessed only by a 2-2 balancer, which is just a single splitter. The drawback is that they get *big*.
+
+### Design principles
+
+Let's think about how to convert a 4-4 balancer into a 3-3. That is done by looping the extra output back into the first splitter. We will use this idea as our base principle:
+
+![Looped 4-4](loop_example_compressed.jpg)
+
+Now, how would we make it so the balancer automatically does this when we block some outputs? Simple, we use priority splitters to auto-loop the items that can't get out. Of course, we need to merge them back with the input too; we use input priority splitters for this, so the main inputs don't slow down when there are items coming. Here is how that might look like:
+
+![Auto-looping 4-4](autoloops_compressed.jpg)
+
+After some testing, we discover a major issue. If the looping items try to enter a belt that's already full, they just back up all the way to the split-off point and block the overflow, putting us back to square one:
+
+![Broken auto-looping 4-4](failed_autoloops_compressed.jpg)
+
+To remedy this, we need to make sure every item on the loop can get to every single one of the loop's exits. This can be done in many ways, but for our needs we're looking for the smallest number of splitters. It just so happens that the most splitter-economical way to do this is with another balancer. Let's add it to the loop:
+
+![Fixed 4-4](secondary_balancer_compressed.jpg)
+
+We now have a fully functional universal 4-4. However, it can be improved. Because of certain balancer mechanics, we don't actually need 4 belts looping back. In general, we need 3 belts less than our balancer is made for. This means we can optimize our 4-4 by compressing the looping belts into 2, and then splitting them again:
+
+![Simpler universal 4-4](simplified_balancer_compressed.jpg)
+
+Now we can shuffle the components around until we get something small, such as this:
+
+![Small universal 4-4](small_balancer_compressed.jpg)
+
+This balancer, while pretty long, is only 6 wide, which means you can fit it in a standard 4 belt bus, and have just enough space to tile them next to each other. It's important to note that this is not my design, although it's based on one of mine. Sadly, I've lost the name of the designer, so I can't give proper credit.
+
+### Advanced design
+
+Now, let's try something bigger: a universal 8-8. We begin by doing the same steps as with our 4-4, except we use 8-8 balancers instead. This is what we get:
+
+![Basic universal 8-8](simple_8-8_compressed.jpg)
+
+Unfortunately, this is an imperfect balancer. Here is one condition which makes it bottleneck:
```

Comment:
What's the difference between this balancer and the 4-4 balancer? If you followed the same steps you should arrive at the same result, no?